pkg/commands/app.go

```diff
@@ -517,6 +518,26 @@ func NewRootfsCommand() *cli.Command {
 	}
 }
 
+// NewSbomCommand is the factory method to add sbom command
+func NewSbomCommand() *cli.Command {
+	return &cli.Command{
+		Name:      "sbom",
+		ArgsUsage: "ARTIFACT",
+		Usage:     "generate sbom for an artifact",
+		Description: fmt.Sprintf(`"ARTIFACT" is the artifact path, trivy will determine the artifact type.
+		To tell trivy the artifact type you can use use the "type":"details" format.
+		Supported types: %s`, artifact.ArtifactTypes),
+		Action: artifact.SbomRun,
+		Flags: []cli.Flag{
+			&formatFlag,
```

Comment:
`table` and `template` don't make sense. We should restrict formats that can be specified.